test/iidm/NodeBreakerVoltageLevelTest.cpp

```diff
@@ -1186,6 +1188,25 @@ BOOST_AUTO_TEST_CASE(TestRemoveVoltageLevelWithInternalConnectionsIssue) {
     POWSYBL_ASSERT_THROW(network.getVoltageLevel("S5 10kV"), PowsyblException, "Unable to find to the identifiable 'S5 10kV'");
 }
 
+BOOST_AUTO_TEST_CASE(NbkComprehensiveErrorMessage) {
+    Network network = powsybl::network::FourSubstationsNodeBreakerFactory::create();
+    BusbarSection& busbarSection = network.getBusbarSection("S1VL2_BBS1");
+    TwoWindingsTransformer& twt = network.getTwoWindingsTransformer("TWT");
+    twt.getPhaseTapChanger().setRegulationTerminal(stdcxx::ref(busbarSection.getTerminal()));
+
+    stdcxx::Properties properties;
+    properties.set(converter::ExportOptions::TOPOLOGY_LEVEL, "NODE_BREAKER");
+
+    // make sure no error is thrown while exporting to NODE_BREAKER topology
+    std::stringstream ss;
+    Network::writeXml("network.xiidm", ss, network, converter::ExportOptions(properties));
+    Network::readXml("network.xiidm", ss);
```

Comment:
You should properly assert that `Network::readXml` and `Network::writeXml` don't throw any exception using `BOOST_CHECK_NO_THROW`. It seems to be possible to write something like that:
```
BOOST_CHECK_NO_THROW({
    std::stringstream ss;
    Network::writeXml("network.xiidm", ss, network, converter::ExportOptions(properties));
    Network::readXml("network.xiidm", ss);
});
```